### PR TITLE
Filter by brand

### PR DESCRIPTION
**Pull request by daoanhhuy26012001**
### Filter by Brand
- [x] Add product filtering based on selected brand types
- [x] Add `type` for object button images
- [x] Using `useState()` to change the status
- [x] Filter by 1 or more brands at once
### Design
- **See more details:** [Link](https://docs.google.com/document/d/1iTrT9cdKrMCjUc6aDLS514gEk7BUFj-m2SCs2wCy2Lw/edit#heading=h.7t5pbcmau6ih)
### Result: [Link](https://drive.google.com/file/d/1Lkr_8NVekhnEuwk7p8p7hbXTwoj5DsUp/view?usp=drive_link)

### CODE REVIEW

File: ReactJS/practice-one/src/App.jsx [hunk 1]
```diff
@@ -28,9 +28,18 @@ function App() {
...
   });
 
+  // Track selected brand type for filtering
+  const [selectedBrandTypes, setSelectedBrandTypes] = useState([]);
+
   const products = useProductData();
   const popoverRef = useRef(null);
 
+  // Filter products based on selected brand type
+  const filteredProducts =
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:38]: The condition too complex. Please make it more easy to read

File: ReactJS/practice-one/src/App.jsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Helpers
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:1]: This one should be a hook. not a helpers

File: ReactJS/practice-one/src/components/Common/Popover/index.jsx [hunk 3]
```diff
@@ -24,15 +39,39 @@ export const Popover = ({ buttonType }) => {
...
 
+  /**
+   * Handle the click event for a brand button.
+   *
+   * @param {string} type - Brand type.
+   */
+  const handleBrandClick = (type) => {
+    if (selectedBrandTypes.includes(type)) {
+      // Brand type is already selected, remove it from the list
+      setSelectedBrandTypes(
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Popover/index.jsx:53]: could you use prevState in useState

File: ReactJS/practice-one/src/App.jsx (hunk 2 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/App.jsx:1]: Done

File: ReactJS/practice-one/src/App.jsx (hunk 1 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/App.jsx:38]: Done

File: ReactJS/practice-one/src/components/Common/Popover/index.jsx (hunk 3 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Popover/index.jsx]: Done

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**